azure_pipeline/src/utils/http.ts

```diff
@@ -0,0 +1,85 @@
+import axios, { AxiosInstance, AxiosRequestConfig, AxiosResponse } from "axios";
+
+
+enum StatusCode {
+  Unauthorized = 401,
+  Forbidden = 403,
+  TooManyRequests = 429,
+  InternalServerError = 500,
+}
+
+const headers: Readonly<Record<string, string | boolean>> = {
+  Accept: "application/json",
+  "Content-Type": "application/json; charset=utf-8",
+  "Access-Control-Allow-Credentials": true,
+  "X-Requested-With": "XMLHttpRequest",
+};
+
+
+class Http {
```

Comment:
it looks like most of the request methods aren't used in these changes, maybe it would be simpler to just have a function to that sends the PUT to update a user, rather than have a general http client initialized to the azure baseUrl. 

Or maybe you are planning on expanding it, and it is pretty simple to wrap axios anyway, so I don't think that it matters all that much. Maybe it is more straightforward to just have the generic client, even if there is slightly more code that is more generic than it looks like it needs to be here if there is only ever going to be the one use in `updateUserConfig`.

Reply:
Good point! I wrote this project last summer and the code definitely worth some improvement. I have made the `http.ts` simpler and clearer. Please let me know if you have more suggestion on the current version.

Reply:
ok cool, the changes look good to me! It would be easy enough to expand it again if needed, but now there isn't the chance that someone could get confused by unused methods.